### PR TITLE
OCT-1730: Improvements for the epoch verifier

### PR DESCRIPTION
**Pull request by kgarbacinski**
## Definition of Done

1. [ ] Acceptance criteria are met.
2. [ ] PR is manually tested before the merge by developer(s).
    - [ ] Happy path is manually checked.
3. [ ] PR is manually tested by QA when their assistance is required (1).
    - [ ] Octant Areas & Test Cases are checked for impact and updated if required (2).
4. [ ] Unit tests are added unless there is a reason to omit them.
5. [ ] Automated tests are added when required.
6. [ ] The code is merged.
7. [ ] Tech documentation is added / updated, reviewed and approved (including mandatory approval by a code owner, should such exist for changed files).
    - [ ] BE: Swagger documentation is updated.
8. [ ] When required by QA:
    - [ ] Deployed to the relevant environment.
    - [ ] Passed system tests.

---

(1) Developer(s) in coordination with QA decide whether it's required. For small tickets introducing small changes QA assistance is most probably not required.

(2) [Octant Areas & Test Cases](https://docs.google.com/spreadsheets/d/1cRe6dxuKJV3a4ZskAwWEPvrFkQm6rEfyUCYwLTYw_Cc).

### CODE REVIEW

File: epoch-verifier/src/data/models.ts [hunk 1]
```diff
@@ -180,3 +208,20 @@ export class EpochUqsImpl implements Deserializable<EpochUqs>{
   }
 }
 
+export class FinalizedSimulationImpl implements Deserializable<FinalizedSimulation>{
+  patronsRewards: bigint;
+  matchedRewards: bigint;
+  projectsRewards: SimulatedReward[];
+
+  from(input: any) {
```
Comment (aziolek) [inline epoch-verifier/src/data/models.ts:216]: Whenever one writes `any` in TS, a tiny penguin dies somewhere.
Not blocking, just sad. ;).

File: epoch-verifier/src/data/models.ts [hunk 2]
```diff
@@ -180,3 +208,20 @@ export class EpochUqsImpl implements Deserializable<EpochUqs>{
...
+    this.patronsRewards = BigInt(input.patronsRewards);
+    this.matchedRewards = BigInt(input.matchedRewards);
+    this.projectsRewards = input.projectsRewards.map((reward: any) => {
```
Comment (aziolek) [inline epoch-verifier/src/data/models.ts:219]: Whenever one writes `any` in TS, a tiny penguin dies somewhere.
Not blocking, just sad. ;).

File: epoch-verifier/src/verifications/utils.ts [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { VerificationResult } from "../runner";
 
 type Predicate<T> = (elem: T) => boolean | VerificationResult
 
-export const SCALE_FACTOR = 10n ** 18n;
+export const SCALE_FACTOR = 10n ** 36n;
```
Comment (aziolek) [inline epoch-verifier/src/verifications/utils.ts:5]: Why the change here?

File: epoch-verifier/src/data/models.ts (hunk 2 above)
Comment (kgarbacinski) [reply epoch-verifier/src/data/models.ts]: let's not kill the poor penguins, fixed

File: epoch-verifier/src/data/models.ts (hunk 1 above)
Comment (kgarbacinski) [reply epoch-verifier/src/data/models.ts]: changed

File: epoch-verifier/src/verifications/utils.ts (hunk 3 above)
Comment (kgarbacinski) [reply epoch-verifier/src/verifications/utils.ts]: I was testing the computation accuracy for different factors and forgot to change it since there's no much difference between 18n and 36n

**Review by aziolek — APPROVED** [review]

**Comment by aziolek** [comment]
It looks like a crash of the runner (Bilbong), not the E2E itself. I suggest rerun:

image
